### PR TITLE
Feat/4661 btc only

### LINKED ISSUE #4661: Install Bitcoin-only firmware directly in Suite
Currently you have to manually upload a file from GitHub. It feels like a non-recommended option. 

**Terminology**
`Universal` - Regular firmware
`Bitcoin-only` - Bitcoin only firmware

### 1. Onboarding
At first part of onboarding include option to switch on Bitcoin firmware.
Screenshot 2022-08-02 at 14 25 44
After clicking on Bitcoin-only, show confirmation window:
Screenshot 2022-08-02 at 14 25 51
Upon confirmation, install:
Screenshot 2022-08-02 at 14 26 02

### 2. Device settings
1. Device settings - add Firmware type section bellow Firmware version section and separate them into a new Firmware category:
Screenshot 2022-08-02 at 14 27 58
2. Show modal after clicking the button
Screenshot 2022-08-02 at 14 28 25

3. Check if you have a seed (no change)
4. Navigate user to access bootloader (no change)
5. Finally install new firmware
Screenshot 2022-08-02 at 14 28 53

### 3. Crypto settings
Show message offering Universal firmware when Bitcoin-only is installed:
Screenshot 2022-08-02 at 14 27 38
Show message offering Bitcoin-only firmware when Universal is installed but only Bitcoin is used:
Screenshot 2022-08-02 at 14 29 35
    
Few more notes in [specification](https://www.notion.so/satoshilabs/Bitcoin-only-FW-in-Suite-a8a83edcfbc74a6ba70dccedd511f8bf).

### PR DESCRIPTION
**Pull request by komret**
- [x] enable Bitcoin-only firmware installation during onboarding
- [x] enable switching between firmware types in Device settings
- [x] display suggestion to install the other type of firmware in Crypto settings 
- [x] enable regtest in Crypto settings for Bitcoin-only firmware
- [x] fix broken Github link when device is in bootloader mode
- [x] Test with an old device requiring intermediary firmware

onboarding
settings
settingssection
settingsmodal
Screenshot 2022-07-22 at 14 50 31
Screenshot 2022-07-22 at 14 52 40

### CODE REVIEW

File: packages/suite/src/actions/firmware/firmwareActions.ts [hunk 1]
```diff
@@ -111,29 +110,40 @@ const firmwareInstall =
                 console.warn(`Installing firmware ${toRelease.release.version}`);
```
Comment (tomasklim) [inline packages/suite/src/actions/firmware/firmwareActions.ts]:
```
console.warn(`Installing ${toBitcoinOnlyFirmware ? "Bitcoin-only" : "Universal"} firmware in version ${toRelease.release.version.join('.')}`);
```

File: packages/suite/src/views/settings/coins/FirmwareTypeSuggestion.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    );
+    const shouldDisplay =
+        device && (bitcoinOnlyFirmware || (!bitcoinOnlyFirmware && onlyBitcoinEnabled));
+
+    if (!shouldDisplay) {
+        return null;
+    }
+
+    return (
+        <Card>
```
Comment (tomasklim) [inline packages/suite/src/views/settings/coins/FirmwareTypeSuggestion.tsx]: To be honest, do we want it there? User was introduced to this problematic during onboarding or when he switched type in Settings>Device

I propose:
- if you really want it there make it at least closable 4ever, same as bech32 account in empty btc account

File: packages/suite/src/actions/firmware/__fixtures__/firmwareActions.ts [hunk 3]
```diff
@@ -48,6 +49,32 @@ export const actions = [
             state: { firmware: { status: 'wait-for-reboot' } },
         },
     },
+    {
```
Comment (tomasklim) [inline packages/suite/src/actions/firmware/__fixtures__/firmwareActions.ts:52]: Would be nice to add test for T1

File: packages/suite/src/actions/firmware/firmwareActions.ts [hunk 4]
```diff
@@ -111,29 +110,40 @@ const firmwareInstall =
...
+            }
+
             analyticsPayload = {
                 toFwVersion: toRelease.release.version.join('.'),
-                toBtcOnly: isBtcOnlyFirmware,
+                toBtcOnly: toBitcoinOnlyFirmware,
             };
 
+            // temporarily save target firmware type so that it can be displayed during installation and restart
+            // the value resets to undefined on firmwareActions.resetReducer() - doing it here would be too early because wee need to keep it during the restart
```
Comment (tomasklim) [inline packages/suite/src/actions/firmware/firmwareActions.ts]:
```suggestion
            // the value resets to undefined on firmwareActions.resetReducer() - doing it here would be too early because we need to keep it during the restart
```

File: packages/suite/src/actions/onboarding/onboardingActions.ts [hunk 5]
```diff
@@ -61,8 +62,15 @@ const goToNextStep = (stepId?: AnyStepId) => (dispatch: Dispatch, getState: GetS
     if (stepId) {
         return dispatch(goToStep(stepId));
     }
+    const { device } = getState().suite;
     const { activeStepId, path } = getState().onboarding;
-    const stepsInPath = steps.filter(step => isStepInPath(step, path));
+    const stepsInPath = steps.filter(step => {
```
Comment (tomasklim) [inline packages/suite/src/actions/onboarding/onboardingActions.ts]: I cannot enable Tor now 😿 

Anyway, I would add it back not only because of Tor but also because you can enable/disable testnet coins and you can change your mind by seeing other coins.

File: packages/suite/src/types/suite/index.ts [hunk 6]
```diff
@@ -145,3 +145,12 @@ export enum TorStatus {
     Disabling = 'Disabling',
     Enabled = 'Enabled',
 }
+
+export enum FirmwareType {
+    BitcoinOnly = 'Bitcoin-only',
+    Universal = 'Universal',
```
Comment (tomasklim) [inline packages/suite/src/types/suite/index.ts:105]: It will confuse us because on some places we call this firmware Regular and in other places Universal :(

File: packages/suite/src/views/settings/device/SettingsDevice.tsx [hunk 7]
```diff
@@ -117,6 +118,7 @@ export const SettingsDevice = () => {
 
                 {!bootloaderMode && (
                     <>
+                        <FirmwareType isDeviceLocked={isDeviceLocked} />
```
Comment (tomasklim) [inline packages/suite/src/views/settings/device/SettingsDevice.tsx]: what about creating a new section `Firmware`. The current `Security` section is kinda too long and firmware is not really `Security`. okay, maybe but no

File: packages/suite/src/components/suite/ModalSwitcher/ForegroundAppModal.tsx [hunk 8]
```diff
@@ -37,10 +37,11 @@ const getForegroundApp = (app: ForegroundAppRoute['app']) => {
 type ForegroundAppModalProps = {
     app: ForegroundAppRoute['app'];
     cancelable: boolean;
+    variant?: ModalVariant;
```
Comment (tomasklim) [inline packages/suite/src/components/suite/ModalSwitcher/ForegroundAppModal.tsx]: As discussed, I would remove `variant` and add a new modal in `ForegroundAppModal`.

File: packages/suite/src/views/onboarding/steps/Firmware/index.tsx [hunk 9]
```diff
@@ -27,6 +27,12 @@ const FirmwareStep = () => {
     } = useFirmware();
     const [cachedDevice, setCachedDevice] = useState<TrezorDevice | undefined>(device);
 
+    const onSuccess = () => {
+        // reset target firmware type
+        resetReducer();
+        goToNextStep();
+    };
+
     // special and hopefully very rare case. this appears when somebody tried to fool user into using a hacked firmware
```
Comment (tomasklim) [inline packages/suite/src/views/onboarding/steps/Firmware/index.tsx:30]: If you have fw, this step does not show which firmware type you have
Screenshot 2022-07-26 at 14 35 18

Maybe it would make sense to provide switch firmware page here?

File: packages/suite/src/views/settings/device/FirmwareType.tsx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    const { anchorRef, shouldHighlight } = useAnchor(SettingsAnchor.FirmwareType);
+
+    if (!device?.features) {
+        return null;
+    }
+
+    const currentFwVersion = getFwVersion(device);
+    const { revision } = device.features;
+    const btcOnly = isBitcoinOnly(device);
+    const githubButtonText = btcOnly ? 'Bitcoin-only' : 'Universal';
```
Comment (tomasklim) [inline packages/suite/src/views/settings/device/FirmwareType.tsx]: Global constant for `Bitcoin-only` and `Universal`

File: packages/suite/src/components/firmware/FirmwareInitial.tsx [hunk 11]
```diff
@@ -80,23 +101,35 @@ const FirmwareInitial = ({
...
+                <Translation
+                    id={
+                        device.firmware === 'none'
+                            ? 'TR_FIRMWARE_SUBHEADING_NONE'
+                            : 'TR_FIRMWARE_SUBHEADING_UNKNOWN'
+                    }
+                    values={{
+                        i: chunks => <i>{chunks}</i>,
+                        button: chunks => (
+                            <StyledButton onClick={() => onInstall(FirmwareType.BitcoinOnly)}>
```
Comment (tomasklim) [inline packages/suite/src/components/firmware/FirmwareInitial.tsx]: Full-featured / Universal / Regular 🤔 
Screenshot 2022-07-26 at 14 54 51

- Text is kinda too long and no one will read it. 
- I don't like the "hidden button" in text. It looks like that it will tell me more about bitcoin-only, not that it will start installation

I propose splitting it into two steps.
1. Firmware installation
- some basic info
2. Firmware type selection
- description of Universal / Regular firmware
- description of bitcoin-only firmware

File: packages/suite/src/support/messages.ts [hunk 12]
```diff
@@ -3072,9 +3093,21 @@ export default defineMessages({
...
-        defaultMessage: 'Your current firmware version is {version}',
+        defaultMessage: 'Your current version is {version}',
         id: 'TR_YOUR_CURRENT_FIRMWARE',
     },
+    TR_YOUR_CURRENT_FIRMWARE_TYPE: {
+        defaultMessage: 'Your firmware type is {version}',
+        id: 'TR_YOUR_CURRENT_FIRMWARE_TYPE',
+    },
+    TR_SWITCH_TO_BITCOIN: {
+        defaultMessage: 'Switch to Bitcoin',
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts:3133]: Switch to Bitcoin-only?

File: packages/suite/src/support/messages.ts [hunk 13]
```diff
@@ -3072,9 +3093,21 @@ export default defineMessages({
         id: 'TR_YOU_WERE_DISCONNECTED_DOT',
     },
     TR_YOUR_CURRENT_FIRMWARE: {
-        defaultMessage: 'Your current firmware version is {version}',
+        defaultMessage: 'Your current version is {version}',
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts:3096]: You are deleting word `firmware` here but in firmware type default message it is there

File: packages/suite/src/support/messages.ts [hunk 14]
```diff
@@ -4409,6 +4442,10 @@ export default defineMessages({
         defaultMessage:
             'Your device is already updated to latest firmware. You may reinstall the firmware if needed.',
     },
+    TR_SWITCH_FIRMWARE_TYPE_DESCRIPTION: {
+        id: 'TR_SWITCH_FIRMWARE_TYPE_DESCRIPTION',
+        defaultMessage: 'You are going to install a different type of firmware.',
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts]: Description of fw types

**Review by tomasklim — CHANGES_REQUESTED** [review]
bitcoin-only firmware setup is working 👍

File: packages/suite/src/views/settings/coins/SettingsCoins.tsx [hunk 15]
```diff
@@ -23,18 +24,9 @@ export const SettingsCoins = () => {
     const { anchorRef: anchorRefTestnetCrypto, shouldHighlight: shouldHighlightTestnetCrypto } =
         useAnchor(SettingsAnchor.TestnetCrypto);
 
-    const { device } = useDevice();
-
     return (
         <StyledSettingsLayout>
-            {device?.connected === false && (
```
Comment (tomasklim) [inline packages/suite/src/views/settings/coins/SettingsCoins.tsx:30]: Is this deleted by accident?

File: packages/suite/src/actions/suite/suiteActions.ts [hunk 16]
```diff
@@ -223,6 +222,19 @@ export const setOnionLinks = (payload: boolean): SuiteAction => ({
     payload,
 });
 
+/**
+ * Triggered by user action in:
+ * - Device Settings
+ * Set `btcOnlyFirmware` field in suite reducer
+ * This reflects firmware type set by the user, not necessarily the one installed
```
Comment (dahaca) [inline packages/suite/src/actions/suite/suiteActions.ts]: If that's the case, what about `switchPreferredFirmwareType`?

File: packages/suite/src/components/firmware/FirmwareOffer.tsx [hunk 17]
```diff
@@ -103,7 +106,11 @@ const FirmwareOffer = ({ device, customFirmware }: Props) => {
     const parsedChangelog =
         !customFirmware && parseFirmwareChangelog(device.features, device.firmwareRelease);
 
-    const bitcoinOnlyVersion = isBitcoinOnly(device) && ' (bitcoin-only)';
+    const getFirmwareTypeName = (btcOnly: boolean) => (btcOnly ? 'Bitcoin-only ' : 'Universal ');
+
+    const currentFirmwareType =
```
Comment (dahaca) [inline packages/suite/src/components/firmware/FirmwareOffer.tsx]: Do you think it would make sense to create a hook with all logic related to BTC-only FW? Check the one I made [for satoshis](https://github.com/trezor/trezor-suite/blob/bfa55be7095f3c4abb86bd8b3e12f97060786f6e/packages/suite/src/hooks/wallet/useBitcoinAmountUnit.ts) for inspiration. :) I am not sure, however, perhaps there is not enough things to abstract in a separate hook.

**Comment by tomasklim** [comment]
Note: Would be nice to add analytics before merging

File: packages/suite/src/actions/firmware/firmwareActions.ts (hunk 1 above)
Comment (komret) [reply packages/suite/src/actions/firmware/firmwareActions.ts]: e1fe2661babdd65547300324d2557ddd0ac053fb

File: packages/suite/src/views/settings/coins/FirmwareTypeSuggestion.tsx (hunk 2 above)
Comment (komret) [reply packages/suite/src/views/settings/coins/FirmwareTypeSuggestion.tsx]: e1fe2661babdd65547300324d2557ddd0ac053fb

File: packages/suite/src/actions/firmware/__fixtures__/firmwareActions.ts (hunk 3 above)
Comment (komret) [reply packages/suite/src/actions/firmware/__fixtures__/firmwareActions.ts:52]: 999de677d1eb05f73ffe6a2246fd9f5b73bdeb4c

File: packages/suite/src/actions/onboarding/onboardingActions.ts (hunk 5 above)
Comment (komret) [reply packages/suite/src/actions/onboarding/onboardingActions.ts]: 82d087647780e1c0bc391d4dc3d9cd4925a3bd9b

File: packages/suite/src/actions/firmware/firmwareActions.ts (hunk 4 above)
Comment (komret) [reply packages/suite/src/actions/firmware/firmwareActions.ts]: fbaa7b408709cc40ccc07d507897ce22d7a57840

File: packages/suite/src/views/onboarding/steps/Firmware/index.tsx (hunk 9 above)
Comment (komret) [reply packages/suite/src/views/onboarding/steps/Firmware/index.tsx:30]: fbaa7b408709cc40ccc07d507897ce22d7a57840

File: packages/suite/src/views/settings/device/FirmwareType.tsx (hunk 10 above)
Comment (komret) [reply packages/suite/src/views/settings/device/FirmwareType.tsx]: fbaa7b408709cc40ccc07d507897ce22d7a57840

File: packages/suite/src/components/firmware/FirmwareInitial.tsx (hunk 11 above)
Comment (komret) [reply packages/suite/src/components/firmware/FirmwareInitial.tsx]: Split between two windows in 999de677d1eb05f73ffe6a2246fd9f5b73bdeb4c

File: packages/suite/src/support/messages.ts (hunk 12 above)
Comment (komret) [reply packages/suite/src/support/messages.ts:3133]: This would make the button longer than others so I would keep this as an exception to the terminology.

File: packages/suite/src/views/settings/coins/SettingsCoins.tsx (hunk 15 above)
Comment (komret) [reply packages/suite/src/views/settings/coins/SettingsCoins.tsx:30]: Yes, reintroduced in e1fe2661babdd65547300324d2557ddd0ac053fb

**Comment by komret** [comment]
> Note: Would be nice to add analytics before merging

This should already be covered by the `firmwareInstall` action which is always used: in onboarding, update and type switch.

```
analyticsPayload = {
                toFwVersion: toRelease.release.version.join('.'),
                toBtcOnly: toBitcoinOnlyFirmware,
            };
```

File: packages/suite/src/support/messages.ts (hunk 14 above)
Comment (komret) [reply packages/suite/src/support/messages.ts]: Improved description in de22367abb28e3d83e315041a0da6957ef04fdac.
Not sure if we should talk about U2F and password manager here or provide links to wiki @hynek-jina 
Screenshot 2022-08-01 at 16 15 45

File: packages/suite/src/actions/firmware/firmwareActions.ts (hunk 1 above)
Comment (tomasklim) [reply packages/suite/src/actions/firmware/firmwareActions.ts]: Probably different commit. I miss `toRelease.release.version.join('.')` the `join` part

Comment (komret) [reply packages/suite/src/actions/firmware/firmwareActions.ts]: Sorry, it's here: fbaa7b408709cc40ccc07d507897ce22d7a57840

**Review by tomasklim — APPROVED** [review]

File: packages/suite/src/views/settings/device/FirmwareVersion.tsx [hunk 18]
```diff
@@ -69,6 +69,16 @@ export const FirmwareVersion = ({ isDeviceLocked }: FirmwareVersionProps) => {
     const currentFwVersion = getFwVersion(device);
     const availableFwVersion = getFwUpdateVersion(device);
     const { revision } = device.features;
+    const githubUrl = GITHUB_FW_COMMIT_URL + revision;
+    const githubButtonIcon = revision ? 'EXTERNAL_LINK' : undefined;
+
+    const handleAction = () => goto('firmware-index', { params: { cancelable: true } });
+
+    const GithubButton = () => (
```
Comment (dahaca) [inline packages/suite/src/views/settings/device/FirmwareVersion.tsx:77]: Why is this needed? 🤔

File: packages/suite/src/views/settings/device/FirmwareVersion.tsx [hunk 19]
```diff
@@ -69,6 +69,16 @@ export const FirmwareVersion = ({ isDeviceLocked }: FirmwareVersionProps) => {
     const currentFwVersion = getFwVersion(device);
     const availableFwVersion = getFwUpdateVersion(device);
     const { revision } = device.features;
+    const githubUrl = GITHUB_FW_COMMIT_URL + revision;
+    const githubButtonIcon = revision ? 'EXTERNAL_LINK' : undefined;
+
+    const handleAction = () => goto('firmware-index', { params: { cancelable: true } });
```
Comment (dahaca) [inline packages/suite/src/views/settings/device/FirmwareVersion.tsx]: `handleUpdate`?

File: packages/suite/src/views/settings/device/FirmwareTypeChange.tsx [hunk 20]
```diff
@@ -0,0 +1,108 @@
...
+            <TextColumn
+                title={<Translation id="TR_FIRMWARE_TYPE" />}
+                description={
+                    currentFwVersion ? (
+                        <Version>
+                            <Translation
+                                id="TR_YOUR_FIRMWARE_TYPE"
+                                values={{
+                                    version: (
+                                        <VersionTooltip content={revision} disabled={!revision}>
```
Comment (dahaca) [inline packages/suite/src/views/settings/device/FirmwareTypeChange.tsx:72]: Could be extracted into the body of the component with `useMemo`, so that the JXS output is clearer.

File: packages/suite/src/views/settings/device/FirmwareTypeChange.tsx [hunk 21]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    const { revision } = device.features;
+    const currentFwVersion = getFwVersion(device);
+    const currentFwType = getFwType(device);
+    const actionButtonId = isBitcoinOnly(device)
+        ? 'TR_SWITCH_TO_UNIVERSAL'
+        : 'TR_SWITCH_TO_BITCOIN';
+
+    const handleAction = () => {
```
Comment (dahaca) [inline packages/suite/src/views/settings/device/FirmwareTypeChange.tsx:51]: `handleUpdate`?

File: packages/suite/src/views/onboarding/steps/Firmware/index.tsx [hunk 22]
```diff
@@ -57,7 +57,7 @@ const FirmwareStep = () => {
                 image="FIRMWARE"
                 heading={<Translation id="TR_FW_INSTALLATION_FAILED" />}
                 description={<Translation id="TOAST_GENERIC_ERROR" values={{ error }} />}
-                innerActions={<RetryButton onClick={firmwareUpdate} />}
+                innerActions={<RetryButton onClick={() => firmwareUpdate()} />}
```
Comment (dahaca) [inline packages/suite/src/views/onboarding/steps/Firmware/index.tsx:60]: Why changing this?

File: packages/suite/src/components/firmware/FirmwareInitial.tsx [hunk 23]
```diff
@@ -8,28 +10,64 @@ import {
...
-    onInstall: () => void;
+    onInstall: (firmwareType?: FirmwareType) => void;
+    switchType?: boolean;
 }
 
 const getDescription = ({
     required,
     standaloneFwUpdate,
     reinstall,
+    switchType,
```
Comment (dahaca) [inline packages/suite/src/components/firmware/FirmwareInitial.tsx:57]: I am a bit confused by that naming, not only in this file. Also some code that follows like:

```
const targetType =
        isBitcoinOnly(device) === !!switchType ? FirmwareType.Universal : FirmwareType.BitcoinOnly;
```

is also unclear because of that 😅 Should it indicate a possibility to switch types or that the type has been switched?

File: packages/suite/src/components/suite/ModalSwitcher/ForegroundAppModal.tsx [hunk 24]
```diff
@@ -11,10 +11,15 @@ import { Version } from '@suite-views/version';
 import { SwitchDevice } from '@suite-components/SwitchDevice';
 import type { ForegroundAppRoute } from '@suite-types';
 
+// would not work if defined directly in the switch
+const FirmwareType = () => <Firmware switchType />;
```
Comment (dahaca) [inline packages/suite/src/components/suite/ModalSwitcher/ForegroundAppModal.tsx]: What about passing the `switchType` prop like `cancelable` is passed further down? Or you like this approach more?

File: packages/suite/src/utils/suite/device.ts [hunk 25]
```diff
@@ -132,6 +132,16 @@ export const isSelectedDevice = (selected?: TrezorDevice | Device, device?: Trez
 
 export const isDeviceInBootloader = (device?: KnownDevice) => !!device?.features.bootloader_mode;
 
+export const isBitcoinOnly = (device: TrezorDevice | Device) => {
```
Comment (dahaca) [inline packages/suite/src/utils/suite/device.ts]: Perhaps it's better to create a general util that would accept a `device` and an array or just one `capability` and return `true` if all passed capabilities are supported? 🤓  

Or did I misunderstand and it checks if that device already has the btc-only FW?

File: packages/suite/src/views/settings/coins/FirmwareTypeSuggestion.tsx [hunk 26]
```diff
@@ -0,0 +1,68 @@
...
+        <StyledCard>
+            <TextColumn
+                description={
+                    <Translation
+                        id={
+                            bitcoinOnlyFirmware
+                                ? 'TR_SETTINGS_COINS_UNIVERSAL_FIRMWARE_SUGGESTION'
+                                : 'TR_SETTINGS_COINS_BITCOIN_FIRMWARE_SUGGESTION'
+                        }
+                        values={{
```
Comment (dahaca) [inline packages/suite/src/views/settings/coins/FirmwareTypeSuggestion.tsx:49]: This could also be placed in the body of `FirmwareTypeSuggestion` to avoid having such a stretched JSX output

File: packages/suite/src/utils/suite/device.ts (hunk 25 above)
Comment (komret) [reply packages/suite/src/utils/suite/device.ts]: This util specifically checks if Bitcoin-only firmware is installed. The way it does so (using features) is not very intuitive, so I would prefer keeping this as a separate function rather than generalize it.

File: packages/suite/src/components/suite/ModalSwitcher/ForegroundAppModal.tsx (hunk 24 above)
Comment (komret) [reply packages/suite/src/components/suite/ModalSwitcher/ForegroundAppModal.tsx]: I prefer this way because (1) the prop is only used in the `Firmware` component so I don't want to clutter the general `ForegroundAppModal`, and (2) the `app` is already checked once in the `getForegroundApp`. But I think either way is fine...

File: packages/suite/src/views/onboarding/steps/Firmware/index.tsx (hunk 22 above)
Comment (komret) [reply packages/suite/src/views/onboarding/steps/Firmware/index.tsx:60]: Because `firmwareUpdate` now receives an argument which is not the default event, so TypeScript complains:
_Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'FirmwareType | undefined'_

File: packages/suite/src/views/settings/device/FirmwareTypeChange.tsx (hunk 20 above)
Comment (komret) [reply packages/suite/src/views/settings/device/FirmwareTypeChange.tsx:72]: I am with you, but this is tricky because of this:

```
if (!device?.features) {
        return null;
    }
```

I cannot place `useMemo` before this because I do not know the revision and I cannot place it after because a hook cannot follow a condition. Do you have any suggestion how to handle this better? Pass `device` as a prop to make sure it is not `undefined`?

File: packages/suite/src/views/settings/device/FirmwareVersion.tsx (hunk 18 above)
Comment (komret) [reply packages/suite/src/views/settings/device/FirmwareVersion.tsx:77]: The `revision` is `undefined` in bootloader mode, so the link does not work. Therefore, I remove the link wrapper when `revision` is not available and assign the button to a variable to avoid duplicate code.

File: packages/suite/src/actions/suite/suiteActions.ts (hunk 16 above)
Comment (komret) [reply packages/suite/src/actions/suite/suiteActions.ts]: This action is gone now, the user's intention is determined by the current route instead and passed directly as a parameter into `firmwareActions.firmwareInstall`.

File: packages/suite/src/components/firmware/FirmwareOffer.tsx (hunk 17 above)
Comment (komret) [reply packages/suite/src/components/firmware/FirmwareOffer.tsx]: I extracted this into `getFwType` util. Besides this, there is only the `isBitcoinOnly` util related to the topic, so I think a hook would be an overkill at this point.

File: packages/suite/src/views/settings/device/FirmwareVersion.tsx (hunk 19 above)
Comment (komret) [reply packages/suite/src/views/settings/device/FirmwareVersion.tsx]: Renamed in 427f67466f36b33deb3c033751dfce922a5e3f00

File: packages/suite/src/actions/firmware/firmwareActions.ts (hunk 1 above)
Comment (tomasklim) [reply packages/suite/src/actions/firmware/firmwareActions.ts]: It is not.

```
            console.warn(
                intermediary
                    ? 'Cannot install latest firmware. Will install intermediary fw instead.'
                    : `Installing ${
                          toBitcoinOnlyFirmware ? FirmwareType.BitcoinOnly : FirmwareType.Universal
                      } firmware ${toRelease.release.version}.`,
            );
```
It misses the `join` part, so there are `,` instead of `.` in console

File: packages/suite/src/components/firmware/FirmwareInitial.tsx (hunk 23 above)
Comment (komret) [reply packages/suite/src/components/firmware/FirmwareInitial.tsx:57]: I made the names and this line more explicit in 427f67466f36b33deb3c033751dfce922a5e3f00

File: packages/suite/src/views/settings/coins/FirmwareTypeSuggestion.tsx (hunk 26 above)
Comment (komret) [reply packages/suite/src/views/settings/coins/FirmwareTypeSuggestion.tsx:49]: 0a51f2bc16851cae212180913925531c861c696b

File: packages/suite/src/actions/firmware/firmwareActions.ts (hunk 1 above)
Comment (komret) [reply packages/suite/src/actions/firmware/firmwareActions.ts]: Ok I see: bd6ecce2e760747026e041fbc6414e47ee7d5519

File: packages/suite/src/utils/suite/device.ts (hunk 25 above)
Comment (dahaca) [reply packages/suite/src/utils/suite/device.ts]: nit: `isDeviceBitcoinOnly`

File: packages/suite/src/components/firmware/FirmwareOffer.tsx (hunk 17 above)
Comment (dahaca) [reply packages/suite/src/components/firmware/FirmwareOffer.tsx]: Yeah, a hook would basically at least have a selector for device and then return stuff like if the FW is bitcoin only. But if you always have access to the device already and you are not accessing it's state only to then use in the util then I think it's ok :)

File: packages/suite/src/views/settings/device/FirmwareTypeChange.tsx (hunk 20 above)
Comment (dahaca) [reply packages/suite/src/views/settings/device/FirmwareTypeChange.tsx:72]: You can use `device?.revision` instead and have the memo before the `return` but it's up to you, it's not so crucial.

**Review by dahaca — APPROVED** [review]
👏👏👏

File: packages/suite/src/utils/suite/device.ts (hunk 25 above)
Comment (komret) [reply packages/suite/src/utils/suite/device.ts]: renamed in b22132bd18adf47a5e711d50663f1d79eca01cca

File: packages/suite/src/components/firmware/FirmwareOffer.tsx (hunk 17 above)
Comment (komret) [reply packages/suite/src/components/firmware/FirmwareOffer.tsx]: `isDeviceBitcoinOnly` sometimes takes cached device as a parameter so a selector could not be used there. However, I used `getFwType` in two more places in b22132bd18adf47a5e711d50663f1d79eca01cca for more uniformity.

File: packages/suite/src/views/settings/device/FirmwareTypeChange.tsx (hunk 20 above)
Comment (komret) [reply packages/suite/src/views/settings/device/FirmwareTypeChange.tsx:72]: There is also one more variable... it is doable but perhaps simpler as it is.

**Comment by komret** [comment]
QA: I updated the issue. Please test the screens I pasted there. The firmware switch should also be tested with a pre-intermediary device.
